views.py

```diff
@@ -93,6 +98,43 @@ def download(request, path, rest_call=False, use_async=True, *args, **kwargs):
                 bag_modified = 'true'
         metadata_dirty = istorage.getAVU(res_root, 'metadata_dirty')
 
+
+
+    if is_zip_download:
+
+        if not path.endswith(".zip"): # requesting folder that needs to be zipped
+            input_path = path.split(res_id)[1]
+            random_hash = random.getrandbits(16)
+            random_hash_path = 'zips/{res_id}/{rand_folder}'.format(res_id=res_id, rand_folder=random_hash)
+            output_path = '{random_hash_path}{path}.zip'.format(random_hash_path=random_hash_path, path=input_path)
+
+            if use_async:
+                task = create_temp_zip.apply_async((res_id, input_path, output_path), countdown=3)
+                delete_zip.apply_async((res_id, random_hash_path), countdown=(20 * 60))  # delete after 20 minutes
```

Comment:
wouldn't it be better to pass in ```output_path``` to ```delete_zip``` so the temp zip can be deleted directly without further path concatenation?

Reply:
I am thinking just err on the safe side, it would be probably beneficial to check zips temp folder and delete them for clean up in ```https://github.com/hydroshare/hydroshare/blob/develop/hs_core/tasks.py#L45``` which is scheduled to run nightly, just in case some exception occurs after temp zips are created and before delete_zip is called, like internet interruption, bad connection, etc. For your reference.

Reply:
Deleting the random_hash_path, we delete everything that was created for the zip.  If I passed the output_path, it would point directly to the zip... leaving empty folders.

Reply:
I agree with setting up a celery task to walk the zips temp folder to delete.  I'll work on adding that.